### PR TITLE
Test for subtract_average and subtract_median

### PR DESCRIPTION
**Pull request by xandrd**
These tests were not part of the new project folder, and the reorganization did not affect it.
Should be safe to merge.

### CODE REVIEW

**Comment by xandrd** [comment]
This merge does not fully address #988. However, it provides the needed test.
The next step is to modify PyTplot.

File: pyspedas/analysis/tests/tests.py [hunk 1]
```diff
@@ -104,6 +110,109 @@ def test_subtract_average(self):
...
+
+        # Additional NaN scenarios
+        store_data('test-nan-all', data={'x': [1., 2., 3., 4.], 'y': [np.NaN, np.NaN, np.NaN, np.NaN]})
+
+        # Check all NaN case
+        result_all = subtract_average('test-nan-all')
+        d_all = get_data(result_all[0])
+        self.assertTrue(np.isnan(d_all[1]).all(), "All NaN input should result in all NaN output.")
+
+    def test01_subtract_average_with_suffix(self):
```
Comment (jameswilburlewis) [inline pyspedas/analysis/tests/tests.py:142]: With some small changes, many tests in this group could be parameterized to take a user-specified function, so they could be reused for other tplot routines that operate on tplot variables one at a time.

**Review by jameswilburlewis — CHANGES_REQUESTED** [review]

File: pyspedas/analysis/tests/tests.py (hunk 1 above)
Comment (xandrd) [reply pyspedas/analysis/tests/tests.py:142]: True. I think when tplot will be merged with pyspedas, we can open an issue to improve testing of pytplot routines.

Comment (jameswilburlewis) [reply pyspedas/analysis/tests/tests.py:142]: The refactoring to make the tests reusable doesn't need to happen right away, so this is good to merge as-is.

**Pull request merged**